libs/ledger-live-common/src/__tests__/api/socket.ts

```diff
@@ -0,0 +1,98 @@
+import {
+  openTransportReplayer,
+  RecordStore,
+} from "@ledgerhq/hw-transport-mocker";
+import { toArray } from "rxjs/operators";
+import scenarios from "./test-helpers/scenarios";
+import WS from "isomorphic-ws";
+
+// import { setEnv } from "@ledgerhq/live-common/env";
+import { createDeviceSocket } from "../../api/socket";
+/**
+ * Both ends of the exchange are mocked in this test file and we are merely testing
+ * the logic that happens in-between.
+ */
+
+jest.mock("isomorphic-ws", () => ({
+  ...jest.requireActual("isomorphic-ws"),
+  __esModule: true,
+  default: jest.fn(),
+}));
+
+describe("Scriptrunner logic", () => {
+  scenarios.forEach(({ describe, device, events }) => {
+    it(describe, async () => {
+      // Generate a transport replayer with the data from the test
+      const transport = await openTransportReplayer(
+        RecordStore.fromString(device)
+      );
+
+      WS.mockImplementation(() => {
+        let msgIndex = 0;
+        const callbacks: { [key: string]: ((any) => void) | undefined } = {
+          onopen: undefined,
+          onerror: undefined,
+          onclose: undefined,
+          onmessage: undefined,
+          send: undefined,
+          close: () => undefined,
+        };
+
+        const onReceiveMessage = (_) => {
+          // Nb get next message and invoke callback
+          msgIndex++;
+          const [callback, payload] = events[msgIndex];
+          if (payload?.includes("e051000000")) {
+            // Need to delay the next apdu to see the allow secure channel event.
+            transport.setArtificialExchangeDelay(1000);
+          }
+          const maybeCallback = callbacks[callback];
+          if (maybeCallback) {
+            maybeCallback({ data: payload });
+          }
+
+          // Check if there's a next event, and if it's a onclose emit it.
```

Comment:
if it's an `onclose` event or if it's an "eager" event ?

Reply:
Not sure i understood the "eager" concept

Reply:
The comment is basically shit, I needed to automatically emit the next event when needed, that's what the concept of eagerness meant. But the comment implies that this only applies to onclose, when it doesn't. I'll change the comment